File: Group Anagrams/Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        indexes = dict()
        output = []

        for s in strs:
            curr = "".join(sorted(s))            
            # print(s, curr)
            if curr in indexes:
                output[indexes[curr]].append(s)
                continue
            
            indexes[curr] = len(output)
            output.append([s])

        return output

File: Group Anagrams/Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;

        for(string s : strs)
        {
            string key = s;
            sort(key.begin(), key.end());

            mp[key].push_back(s);
        }

        vector<vector<string>> output;

        for(auto pair : mp)
        {
            output.push_back(pair.second);
        }
        return output;
    }
};